README.md

```diff
@@ -51,28 +49,55 @@ import logging
 
 
 # 1. Define a function that accepts hyperparameters and computes the validation error
-def run_pipeline(hyperparameter_a: float, hyperparameter_b: int):
-    validation_error = -hyperparameter_a * hyperparameter_b
-    return validation_error
+def run_pipeline(
+    hyperparameter_a: float, hyperparameter_b: int, architecture_parameter: str
+) -> dict:
+    # Create your model
+    model = MyModel(architecture_parameter)
+
+    # Train and evaluate the model with your training pipeline
+    validation_error, test_error = train_and_eval(
+        model, hyperparameter_a, hyperparameter_b
+    )
+
+    return {  # dict or float(validation error)
+        "loss": validation_error,
+        "info_dict": {
+            "test_error": test_error
+            # + Other metrics
+        },
+    }
 
 
 # 2. Define a search space of hyperparameters; use the same names as in run_pipeline
 pipeline_space = dict(
-    hyperparameter_a=neps.FloatParameter(lower=0, upper=1),
-    hyperparameter_b=neps.IntegerParameter(lower=1, upper=100),
+    hyperparameter_b=neps.IntegerParameter(
+        lower=1, upper=42, is_fidelity=True
+    ),  # Mark 'is_fidelity' as true for a multi-fidelity approach.
+    hyperparameter_a=neps.FloatParameter(
+        lower=0.001, upper=0.1, log=True
+    ),  # If True, the search space is sampled in log space.
+    architecture_parameter=neps.CategoricalParameter(
+        ["option_a", "option_b", "option_c"]
+    ),
 )
 
-# 3. Call neps.run to optimize run_pipeline over pipeline_space
-logging.basicConfig(level=logging.INFO)
-neps.run(
-    run_pipeline=run_pipeline,
-    pipeline_space=pipeline_space,
-    root_directory="usage_example",
-    max_evaluations_total=5,
-)
+if __name__ == "__main__":
+    # 3. Run the NePS optimization
+    logging.basicConfig(level=logging.INFO)
+    neps.run(
+        run_pipeline=run_pipeline,
+        pipeline_space=pipeline_space,
+        root_directory="path/to/save/results",  # Replace with the actual path.
+        max_evaluations_total=100,
+        searcher="hyperband"  # Optional specifies the search strategy,
+        # otherwise NePs decides based on your data.
+    )
 ```
 
-For more details and features please have a look at our [documentation](https://automl.github.io/neps/latest/) and [examples](neps_examples).
+## Documentation
+
+For more details and features please have a look at our [documentation](https://automl.github.io/neps/latest/) and [examples](neps_examples)
 
 ## Analysing runs
 
```

Comment:
Once the citation block is added, can we merge this?

Reply:
Citations are now added [here](https://github.com/automl/neps/blob/readme_changes_danrgll/docs/citations.md).